### DISCUSSION TITLE
Context Trees | Rewrite

### DISCUSSION THREAD

**Discussion by aniketfuryrocks** [gxi-rs/gxi#24] [Announcements]
After switching to the observable pattern I noticed a particular trend in the paradigm of UI frameworks, i.e doing more than what is required for just state management. The functional observable pattern that was being developed for the gxi project, itself has a lot of problems concerning memory leaks and extended scopes or lifetime. I myself had some concerns while developing it, but it was the most optimal solution at the time.

Recently I came up with the idea of `context-trees` which not only solves the problem with the observable pattern but also fix the lifetime extension problem. Using the `context-trees` I am able to directly bind `state`, `mutation view node` and `consumer node`. Therefore, in theory, this new pattern will produce the same performance as any vanilla implementation while making complex features like live `tree-shaking` possible.

This is the 8th and hopefully the final re-write of this project. I have started working on it, but it will for sure take a lot of time.

**Comment by SamuraiCrow** [comment]
I'm glad you're taking the time to get this right.  To become a standard function of a programming language takes careful consideration.
